java/client/src/org/openqa/selenium/remote/AddRotatable.java

```diff
@@ -19,28 +19,61 @@
 
 import com.google.common.collect.ImmutableMap;
 
+import org.openqa.selenium.DeviceRotation;
 import org.openqa.selenium.Rotatable;
 import org.openqa.selenium.ScreenOrientation;
 
 import java.lang.reflect.Method;
 
 public class AddRotatable implements AugmenterProvider {
 
+  private enum RotatableMethod {
+    ROTATE("rotate"),
+    GET_ORIENTATION("getOrientation"),
+    GET_ROTATION("rotation");
+      
+    private String methodName;
+      
+    private RotatableMethod(String methodName) {
+      this.methodName = methodName;
+    }
+    
+    public String toString() {
+      return this.methodName;
+    }
+  }
+    
   public Class<?> getDescribedInterface() {
     return Rotatable.class;
   }
 
   public InterfaceImplementation getImplementation(Object value) {
     return new InterfaceImplementation() {
       public Object invoke(ExecuteMethod executeMethod, Object self, Method method, Object... args) {
-        if ("rotate".equals(method.getName())) {
-          return executeMethod.execute(DriverCommand.SET_SCREEN_ORIENTATION,
-              ImmutableMap.of("orientation", args[0]));
-        } else if ("getOrientation".equals(method.getName())) {
-          return ScreenOrientation.valueOf((String) executeMethod.execute(
-              DriverCommand.GET_SCREEN_ORIENTATION, null));
+        RotatableMethod m = RotatableMethod.valueOf(method.getName());
+        Object methodImp;
+        switch(m) {
+          case ROTATE:
+            String cmd = "";
+            if (args[0] instanceof ScreenOrientation) {
+              cmd = DriverCommand.SET_SCREEN_ORIENTATION;
+            } else if (args[0] instanceof DeviceRotation) {
+              cmd = DriverCommand.SET_SCREEN_ROTATION;
+            } else {
+              throw new IllegalArgumentException("rotate parameter must be either of type 'ScreenOrientation' or 'DeviceRotation'");
+            }
+            methodImp = executeMethod.execute(cmd, ImmutableMap.of("orientation", args[0]));
```

Comment:
i didn't think that setting the Rotation with DeviceRotation should send the payload with "orientation" as the key in the json object.

https://github.com/SeleniumHQ/mobile-spec/blob/master/spec-draft.md#devicerotation

If we're strictly following what is said there, it should be "parameters" in that case... but that was written when we thought the w3c spec was also going to do that, but it's not. I'd recommend that you just do:

    executeMethod.execute(DriverCommand.SET_SCREEN_ROTATION, ((DeviceRotation)args[0]).parameters());

for the case of when it is DeviceRotation... (and then we should update the spec :) )